src/Microsoft.HttpRepl/HttpState.cs

```diff
@@ -108,6 +119,18 @@ public static Uri GetEffectivePath(Uri baseAddress, string pathSections, string
             AppendQueryToBuilder(builder, baseAndPathQuery);
             AppendQueryToBuilder(builder, commandQuery);
 
+            if (queryString != null)
+            {
+                
+                foreach (KeyValuePair<string, IEnumerable<string>> tuple in queryString)
+                {
+                    foreach (var singleValue in tuple.Value)
+                    {
+                        AppendQueryToBuilder(builder, $"{tuple.Key}={WebUtility.UrlEncode(singleValue)}");
```

Comment:
```suggestion
                        AppendQueryToBuilder(builder, $"{WebUtility.UrlEncode(tuple.Key)}={WebUtility.UrlEncode(singleValue)}");
```

According to [“application/x-www-form-urlencoded “](https://www.w3.org/TR/html401/interact/forms.html#didx-applicationx-www-form-urlencoded):

> 1. Control names and values are escaped.

It's also best to err on the safe side. I'd also recommend adding a test for this.

Also, any considerations for the following?

> Additionally, the `allowReserved` keyword specifies whether the reserved characters `:/?#[]@!$&'()*+,;=` in parameter values are allowed to be sent as they are, or should be percent-encoded. By default, `allowReserved` is `false`, and reserved characters are percent-encoded. For example, `/` is encoded as `%2F` (or `%2f`), so that the parameter value `quotes/h2g2.txt` will be sent as `quotes%2Fh2g2.txt`

Reply:
re: `allowReserved`:  I'd be curious to know how often allowReserved is true on query parameters (a quick [search of APIs.guru's OpenAPI Directory](https://github.com/APIs-guru/openapi-directory/search?q=allowReserved) turned up only 1 result, though I doubt that's fully representative. My instinct says we should table that and add an issue to track supporting it later, in large part because that's a per-endpoint decision and the parameter definitions aren't available at the time this code is being called, thus making it a larger change to support it. Seems like a good candidate for its own issue/PR.

Reply:
@ygalbel I'll add a tracking issue for the `allowReserved` portion of this discussion and mention it here when I do. Then I think you just need the change @atifaziz mentioned suggested to encode the name as well as the value. A test would be good too :)